crates/symbolicator-sources/src/sources.rs

```diff
@@ -134,13 +134,34 @@ pub struct SourceFilters {
     /// If a debug image does not contain any path information it will be treated like an image
     /// whose path doesn't match any pattern.
     pub path_patterns: Vec<Glob>,
+
+    /// Whether this server requires a "debug checksum" to download portable PDB files.
+    #[serde(default)]
+    pub requires_checksum: bool,
 }
 
 impl SourceFilters {
     /// Whether the [`ObjectId`] / [`FileType`] combination is allowed on this source.
     pub fn is_allowed(&self, object_id: &ObjectId, filetype: FileType) -> bool {
-        (self.filetypes.is_empty() || self.filetypes.contains(&filetype))
-            && paths::matches_path_patterns(object_id, &self.path_patterns)
+        // Checks if a checksum is required for portable PDB downloads
+        if self.requires_checksum
+            && filetype == FileType::PortablePdb
```

Comment:
Do we need to check the filetype explicitly here if the source already contains the information whether a checksum is required or not?

Reply:
My thinking here was that these checksums only exist of portable PDB files in the first place. The only example we have of this is NuGet, but _hypothetically_ there could be another source that requires checksums for portable PDB files but not for other files.

It probably doesn't make sense to worry about this; I'll remove the filetype check. If such a source ever pops up we can fix it.